src/storage/wal_replayer.cpp

```diff
@@ -536,7 +521,7 @@ BMFileHandle* WALReplayer::getVersionedFileHandleIfWALVersionAndBMShouldBeCleare
             }
         }
         default: {
-            assert(false);
+            return nullptr;
```

Comment:
Generally `assert(false)` is not a good principle. It can only be triggered during debug mode, and the compiler is not able to detect it and throw warnings. We're shifting to throw NotImplementedException. Changed to throw exception here.